src/js/utils/html-utils.js

```diff
@@ -0,0 +1,115 @@
+         * replaced by the parameter value of the same name.
+         *
+         * Parameter values will be rendered using their toString methods and then
+         * HTML-escaped. The only exception is that instances of the class HTML
+         * are rendered without escaping as their contract declares that they are
+         * already valid HTML.
+         *
+         * Example:
+         *   HtmlUtils.interpolate(
+         *       'You are enrolling in {spanStart}{courseName}{spanEnd}',
+         *       {
+     *           courseName: 'Rock & Roll 101',
+     *           spanStart: HtmlUtils.HTML('<span class="course-title">'),
+     *           spanEnd: HtmlUtils.HTML('</span>')
+     *       }
+         *   );
+         *
+         * returns:
+         *   'You are enrolling in <span class="course-title">Rock &amp; Roll 101</span>'
+         *
+         * Note: typically the formatString will need to be internationalized, in which
+         * case it will be wrapped with a call to an i18n lookup function. In Django,
+         * this would look like:
+         *
+         *   HtmlUtils.interpolate(
+         *       gettext('You are enrolling in {spanStart}{courseName}{spanEnd}'),
+         *       ...
+         *   );
+         *
+         * @param {string} formatString The string to be interpolated.
+         * @param {Object} parameters An optional set of parameters to the template.
+         * @returns {HTMLSnippet} The resulting HTML snippet that has been escaped.
+         */
+        interpolate = function(formatString, parameters) {
+            var escapedParameters = escapeInterpolateParameters(parameters),
+                result = _.escape(formatString).replace(/{\w+}/g,
+                    function(parameter) {
+                        var parameterName = parameter.slice(1,-1);
+                        return String(escapedParameters[parameterName]);
+                    });
+            return HTML(result);
+        };
+
+        /**
+         * Returns an object with each of the parameters appropriately escaped.
+         *
+         * Note: if a parameter value is an HTMLSnippet then it does not need
+         * to be further escaped, but otherwise it will be escaped before being
+         * included in the result.
+         *
+         * @param {Object} parameters The parameter values to be interpolated.
+         * @returns {Object} An object with each parameter value escaped as necessary.
+         */
+        escapeInterpolateParameters = function(parameters) {
+            return _.mapObject(parameters, function(value) {
+                if (value instanceof HtmlSnippet) {
+                    return value;
+                } else {
+                    return _.escape(value);
+                }
+            });
+        };
+
+        return {
+            interpolate: interpolate,
+            HTML: HTML,
+            HtmlSnippet: HtmlSnippet
```

Comment:
Good point, @dan-f. I think my new method ``HtmlUtils.escape`` does exactly what @robrap wanted, in that it will only escape if necessary. I suppose you could now argue that that means that ``HtmlSnippet`` can become internal again. I'm tempted to leave it public so that the documentation is clearer, and because I could imagine there being other potential usages.